implementations/rust/ockam/ockam_api/src/minicbor_url.rs

```diff
@@ -0,0 +1,64 @@
+use minicbor::{decode, encode, Decode, Decoder, Encode, Encoder};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+use url::Url as RegularUrl;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Url(RegularUrl);
```

Comment:
Could you please add a comment here to explain why we need to go through this gymnastics with a newtype?